### ISSUE TITLE
🚀 Feature: Prisma Integration (connector)

### ISSUE THREAD

**Issue by hmmhmmhm** [appwrite/appwrite#3005]
### 🔖 Feature description

Prisma is a ORM that has very strong automatic migration capabilities
and strong db accessibility. (Very convenient to build some raw query)

Prisma will also be able to support more complex DB features if Prisma is made available with Appwrite.
If we can make Prisma work with appwrite, It should be able to support much complex DB functions.

Regardless of Prisma's migration, being able to using some
prisma query build system will be a huge advantage.

Prisma already supports Supbase, Keystone, and many other frameworks.
And it is being used to increase their DB utilization.
( Supabase Prisma Guide: https://supabase.com/docs/guides/integrations/prisma )
( Keystone Prisma Guide: https://v5.keystonejs.com/guides/prisma )

### 🎤 Pitch

One of the main reasons why Appwrite is not applicable to main production
is that it does not take full advantage of the powerful features of DB.

With Appwrite, it is also impossible to inject data directly into the DB without going through the Appwrite API.

It requires need to change the Appwrite system,
which does not currently recommend direct query of mutation to DB.

( And it might be caused by redis caching.
I think we should provide some button to disable redis caching,
etc. that adversely affects direct data muate to db. )

If we can use proven ORM with Appwrite, we will be able to significantly
increase the number of Appwrite based productions.

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by brandonroberts** [comment]
Thanks, @hmmhmmhm. From a feature perspective, Prisma definitely could provide a powerful set of tools for managing, migrating, and querying data from Appwrite. I think myself and others in the community believe that Appwrite is suitable and proven to be used in production today also while being flexible enough to be integrated with a tool like Prisma.

I think there's more exploration to be done on the integration side, as there is definitely a balance to be had in terms of what most users of Appwrite would use vs what's available out of the box in terms of database access. We are always looking to make development easier and faster with the Appwrite API, in addition to supporting more advanced use cases. And of course the community is always able to dig into the code and explore these types of integrations also.

We will definitely keep this on our radar moving forward.

**Comment by FranciscoKloganB** [comment]
Any news on these? I was really keen on subscribing AppWrite cloud, been waiting for it for months, but then I discovered I "do not" have direct access to my database.

I was subscribbing AppWrite cloud mainly for "managed user auth", "no infrastructure worries", "7 day log retention", "image transformation", "storage" all in one. But honestly, I really want to have full control of my model.

I understand AppWrite provides alot of power mechanisms through abstraction. However, personally, I want that abstractions with other things. Database querying is bread and butter, I do woud not want to "learn" something new, which for my use-case is trivially solved with simple queries, that happens to vendor locks me!

**Comment by joelmce** [comment]
Second this.

**Comment by imskanand** [comment]
What is the update regarding this integration `Appwrite + Prisma` ? is it possible in the near future to use `prisma` or with some other orm like `Drizzle`?

**Comment by amazingnerd** [comment]
Is there a solution yet everybody?

**Comment by camstuart** [comment]
This would be just magic. An SDK agnostic tool for describing and managing the 'appwrite infra' not just databases would be amazing. But certainly starting at the db makes a lot of sense.

Ideally we should be able to describe structure in a common DSL. I just wrote a bunch of python to set up my database. But there is a lot more todo. Many more collections, relationships, indexes and permissions. So I started abstracting. However it feels like I am re-inventing the wheel.

Interestingly, Prisma has found it's way into the python ecosystem: https://prisma-client-py.readthedocs.io/en/stable/

**Comment by dwissaaj** [comment]
Is there any solution about this? I dont really grasp good side of appwrite SDK. Prefer prisma and directly to their database like prisma + supabase

**Comment by cogscides** [comment]
+1

**Comment by tabzzz1** [comment]
+1

**Comment by DeepBhat** [comment]
Having an integration with Prisma would be great, even if it is just to get migrations and type safe queries.

**Comment by LukeHagar** [comment]
Agreed here, writing a script to template out a Collection is not ideal.

**Comment by AndreaDev237** [comment]
+1

**Comment by humayunkabir** [comment]
+1

**Comment by hmmhmmhm** [comment]
Even after three years, I still want this feature. And many developers in the community still want it too. @eldadfux would you be able to take another look at this feature..? 🥺

**Comment by mmaryo** [comment]
+1

**Comment by ebenezerdon** [comment]
Hey everyone, thanks so much for all the feedback and discussion around this.

At the moment, Prisma integration isn't something we're planning to support. Appwrite is built as an API-first platform to offer a consistent and secure way to build apps, and direct ORM access doesn't quite fit with that model.

We totally get why Prisma is appealing, but adding it wouldn't really address the deeper concerns being raised. Instead, we're putting our focus on expanding Appwrite's built-in database features to better support production use cases.

Really appreciate the input. It's helping us move in the right direction.

**Comment by dwissaaj** [comment]
kinda make sense, whats the point of have appwrite sdk if you working again for prisma 😂

**Comment by cccabdalla** [comment]
I noticed this request has been around for years and keeps coming up as Prisma becomes more common.

Is there any updated stance or roadmap item regarding Prisma integration with Appwrite?

Many of us would appreciate clarity, even if it’s a “not planned” decision.

**Comment by BretRen** [comment]
I don't know what Appwrite's internal plans are, but as of now (2026), Appwrite doesn't seem to be prioritizing this as a core feature. However, someone has created [Appwrite ORM](https://appwrite-orm.online/).

**Comment by cccabdalla** [comment]
> I don't know what Appwrite's internal plans are, but as of now (2026), Appwrite doesn't seem to be prioritizing this as a core feature. However, someone has created [Appwrite ORM](https://appwrite-orm.online/).
> 
> 


Thank you for sharing this Appwrite ORM link. Will look into.

**Comment by AnuragKrVerma** [comment]
[Appwrite ORM](https://appwrite-orm.online/) interesting but  i am still waiting for Prisma ORM & Drizzle ORM Integration
